File: lib/models/patient_model.dart
import 'gender_model.dart';

class PatientModel {
  final String firstName;
  final String lastName;
  final String email;
  final String password;
  final String confirmPassword;
  final Gender gender;
  final String phoneNumber;
  final DateTime? birthday;
  PatientModel({
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.password,
    required this.confirmPassword,
    required this.gender,
    required this.phoneNumber,
    this.birthday,
  });

  String genderToString(Gender gender) {
    switch (gender) {
      case Gender.male:
        return 'Male';
      case Gender.female:
        return 'Female';
    }
  }

  Map<String, dynamic> toJson() {
    return {
      'pFirstName': firstName,
      'pLastName': lastName,
      'email': email,
      'password': password,
      'confirmPassword': confirmPassword,
      'gender': genderToString(gender),
      'phone': phoneNumber,
      'birthday': birthday?.toIso8601String(),
    };
  }
}

File: backend/DB/models/patient.model.js
import mongoose, { Schema , model} from "mongoose"


//schema
const PatientSchema = new Schema({
    pFirstName : {
        type: String,
        required:true,
        min:3,
        max:20 
    },
    pLastName : {
        type: String,
        required:true,
        min:3,
        max:20 
    },
    email :{
        type: String,
        unique: true,
        required:true,
        lowercase:true
    },
    password: {
        type: String,
        required: true 
    },
    gender: {
        type: String,
        enum: ["Male","Female"]
    },
    phone:{
        type: String
    },
    birthday:Date,
    
    status: {
        type: String,
        enum: ["online","offline"],
        default:"offline"
    },
    isConfirmed:{
        type:Boolean,
        default:false
    },forgetCode:String,
    activationCode:String,
    role: {
        type: String,
        default: 'patient', // Default role for patients
        
    },
    image: [{ type: mongoose.Schema.Types.ObjectId, ref: 'image' }],
    report: [{ type: mongoose.Schema.Types.ObjectId, ref: 'report' }],
    chosenDoctor:String,
    
    

},{timestamps : true})

//model
export const Patient = mongoose.models.Patient || model("Patient" , PatientSchema)

